crates/ra_assists/src/assists/raw_string.rs

```diff
@@ -28,7 +28,25 @@ pub(crate) fn make_raw_string(mut ctx: AssistCtx<impl HirDatabase>) -> Option<As
         if error.is_err() {
             eprintln!("Error unescaping string");
         } else {
-            edit.replace(literal.syntax().text_range(), format!("r#\"{}\"#", unescaped));
+            let mut max_hash_streak = 0;
+            unescaped.chars().fold(0, |acc, c| {
+                if c == '#' {
+                    acc + 1
+                } else {
+                    if acc > max_hash_streak {
+                        max_hash_streak = acc;
```

Comment:
To my mind this doesn't account for group of #s which is right before string end.

E.g., for string "########" it won't try to update max_hash_streak at all

Reply:
What about `"#abc ###"`? The last 3 hashes doesn't count? I didn't know that :D

Reply:
I think this can be implemented much simpler, without any folds: look for `"#` sub strings using `find`, then count hashes using `.take_while(|it| it == '#').count()`. This also can be extracted into a separate function with simple interface, which can be unit-tested:

```
fn count_hashes(s: &str) -> usize
```

Reply:
Note that `unescaped` is only the internal string, without leading and ending `"`. Does your `"` in `"#` is the starting one? or it is like:
```
"my string\nto unescape\"# has hash"
```

ok for the separate function to unit test